packages/auth-remix/readme.md

```diff
@@ -0,0 +1,157 @@
+
+   Then you just need to configure the `onBuiltinUICallback` route handler to define what to do once the builtin ui redirects back to your app, and place a link to the builtin UI url returned by `auth.getBuiltinUIUrl()` somewhere in your app.
+
+   **Custom UI**
+
+   To help with implementing your own custom auth UI, the `auth` object has a number of methods you can use:
+
+   - `emailPasswordSignUp`
+   - `emailPasswordSignIn`
+   - `emailPasswordResendVerificationEmail`
+   - `emailPasswordSendPasswordResetEmail`
+   - `emailPasswordResetPassword`
+   - `signout`
+   - `isPasswordResetTokenValid(resetToken: string)`: Checks if a password reset token is still valid.
+   - `getOAuthUrl(providerName: string)`: This method takes the name of an OAuth provider (make sure you configure providers you need in the auth ext config first using CLI or EdgeDB UI) and returns a link that will initiate the OAuth sign in flow for that provider. You will also need to configure the `onOAuthCallback` auth route handler.
+
+## Usage
+
+Now you have auth all configured and user's can signin/signup/etc. you can use the `auth.getSession()` method in your app pages to retrieve an `AuthSession` object. This session object allows you to check if the user is currently logged in with the `isLoggedIn` method, and also provides a `Client` object automatically configured with the `ext::auth::client_token` global, so you can run queries using the `ext::auth::ClientTokenIdentity` of the currently signed in user.
+
+```ts
+// app/routes/_index.tsx
+import type { LoaderFunctionArgs } from "@remix-run/node";
+import { json } from "@remix-run/node";
+import { useLoaderData, Link } from "@remix-run/react";
+import { client, auth } from "~/services/auth.server";
+import { transformSearchParams } from "~/utils";
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  const builtinUIEnabled = await client.queryRequiredSingle<boolean>(
+    `select exists ext::auth::UIConfig`
+  );
+
+  const session = auth.getSession(request);
+  const isLoggedIn = await session.isLoggedIn();
+  const username = isLoggedIn
+    ? (await session.client.querySingle<string>(
+        `select global currentUser.name`
+      )) || ""
+    : "";
+
+  let params = new URL(request.url).searchParams;
+
+  return json({
+    builtinUIEnabled,
+    builtinUIUrl: auth.getBuiltinUIUrl(),
+    isLoggedIn,
+    username,
+    params: transformSearchParams(params),
+  });
+};
+
+export default function Index() {
+  const { builtinUIUrl, isLoggedIn, username } = useLoaderData<typeof loader>();
+
+  return (
+    <main>
+      <h1>Home</h1>
+      {isLoggedIn ? (
+        <>
+          <h2>
+            You are logged in as<span>{username}</span>
```

Comment:
```suggestion
You are logged in{username ? ` as <span>${username}</span>` : ""
```

—?

Reply:
Hey @scotttrinh, I think it's good to make username required and to fallback to null, but just to be sure I got this, the end result in your suggestion is the same for empty string or null? So using null is more like a good practice preference or for someone debugging, we show here to the user `You are logged in ` in case username is null but somehow user is logged in.

Reply:
Yeah, the result is the same, I was just handling the case where the name was not required.

If I was doing this in a real app, I would make some discriminated union that helped the type checker know that if a user was logged in, the name is not null, but not necessary for this example I think.

Reply:
Btw, the suggestion was initially to fix the missing space before the name but I ended up also suggesting changes for optional name.